temporalio/worker/workflow_instance.py

```diff
@@ -0,0 +1,1396 @@
+                self._apply(job)
+
+            # Run one iteration of the loop
+            self._run_once()
+
+        # Return pending commands
+        return self._pending_commands
+
+    def _apply(
+        self, job: temporalio.bridge.proto.workflow_activation.WorkflowActivationJob
+    ) -> None:
+        if job.HasField("cancel_workflow"):
+            self._apply_cancel_workflow(job.cancel_workflow)
+        elif job.HasField("fire_timer"):
+            self._apply_fire_timer(job.fire_timer)
+        elif job.HasField("query_workflow"):
+            self._apply_query_workflow(job.query_workflow)
+        elif job.HasField("notify_has_patch"):
+            # TODO(cretz): This
+            pass
+        elif job.HasField("remove_from_cache"):
+            # Ignore, handled externally
+            pass
+        elif job.HasField("resolve_activity"):
+            self._apply_resolve_activity(job.resolve_activity)
+        elif job.HasField("resolve_child_workflow_execution"):
+            self._apply_resolve_child_workflow_execution(
+                job.resolve_child_workflow_execution
+            )
+        elif job.HasField("resolve_child_workflow_execution_start"):
+            self._apply_resolve_child_workflow_execution_start(
+                job.resolve_child_workflow_execution_start
+            )
+        elif job.HasField("resolve_request_cancel_external_workflow"):
+            # TODO(cretz): This
+            pass
+        elif job.HasField("resolve_signal_external_workflow"):
+            # TODO(cretz): This
+            pass
+        elif job.HasField("signal_workflow"):
+            self._apply_signal_workflow(job.signal_workflow)
+        elif job.HasField("start_workflow"):
+            self._apply_start_workflow(job.start_workflow)
+        elif job.HasField("update_random_seed"):
+            # TODO(cretz): This
+            pass
+        else:
+            raise RuntimeError(f"Unrecognized job: {job.WhichOneof('variant')}")
+
+    def _apply_cancel_workflow(
+        self, job: temporalio.bridge.proto.workflow_activation.CancelWorkflow
+    ) -> None:
+        # TODO(cretz): Details or cancel message or whatever?
+        if self._primary_task:
+            self._primary_task.cancel()
+
+    def _apply_fire_timer(
+        self, job: temporalio.bridge.proto.workflow_activation.FireTimer
+    ) -> None:
+        handle = self._pending_timers.pop(job.seq, None)
+        if not handle:
+            raise RuntimeError(f"Failed finding timer handle for sequence {job.seq}")
+        # Mark cancelled so the cancel() from things like asyncio.sleep() don't
+        # invoke _timer_handle_cancelled
+        handle._cancelled = True
+        self._ready.append(handle)
+
+    def _apply_query_workflow(
+        self, job: temporalio.bridge.proto.workflow_activation.QueryWorkflow
+    ) -> None:
+        # Async call to run on the scheduler thread
+        async def run_query(input: HandleQueryInput) -> None:
```

Comment:
There's an open issue to update time before the query in sdk-features, might as well do that here already (can wait for another PR but should be tracked): https://github.com/temporalio/sdk-features/issues/31

Reply:
Hrmm, I think I disagree with that issue. I commented there.

Reply:
Commented on the issue, not blocking this PR.